Wikipedia/View Controllers/Nearby/NearbyViewController.m

```diff
@@ -122,12 +113,50 @@ - (BOOL)prefersStatusBarHidden {
     return NO;
 }
 
-- (void)viewWillDisappear:(BOOL)animated {
-    [self.locationManager stopUpdatingLocation];
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+
+    self.skipStartUpdates = NO;
+    // Only begin updates if by the time the startUpdates selector below is performed
+    // self.skipStartUpdates is still NO - lets us add the nearby view controller
+    // somewhere and immediately cover it up without it starting to update locations
+    // and headings.
+    [self performSelector:@selector(startUpdates) withObject:nil afterDelay:0.5];
+}
+
+- (void)startUpdates {
+    if (!self.skipStartUpdates) {
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            // Needed by iOS 8.
+            SEL selector = NSSelectorFromString(@"requestWhenInUseAuthorization");
+            if ([self.locationManager respondsToSelector:selector]) {
+                NSInvocation* invocation =
```

Comment:
Wow not sure where I got that old code for doing this... but wow... sure was silly :)